drivers/spi/spi_ll_stm32.c

```diff
@@ -90,9 +99,26 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 	u16_t rx_frame;
 
 	tx_frame = spi_stm32_next_tx(data);
+
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	/* STM32MP1 has some register name modifications :
+	 * LL_SPI_IsActiveFlag_TXP <-> LL_SPI_IsActiveFlag_TXE
+	 * LL_SPI_IsActiveFlag_RXP <-> LL_SPI_IsActiveFlag_RXNE
+	 * Also, if the device is the SPI master, we need to enable the start of
+	 * the transfer with LL_SPI_StartMasterTransfer(spi)
+	 */
+	while (!LL_SPI_IsActiveFlag_TXP(spi)) {
+		/* NOP if not enough space for a packet*/
+	}
+	if (LL_SPI_IsEnabled(spi) && LL_SPI_GetMode(spi)) {
```

Comment:
`LL_SPI_GetMode(spi)` likely doesn't return true/false, please test against `LL_SPI_MODE_MASTER`.
OTOH, I think `LL_SPI_IsEnabled` will logically return true here.

Reply:
`LL_SPI_IsEnabled` return either 1 (true) or 0 (false)
`LL_SPI_GetMode(spi)` return either :
```
#define LL_SPI_MODE_MASTER                      (SPI_CFG2_MASTER)
#define LL_SPI_MODE_SLAVE                          (0x00000000UL)
```

I agree this might not be super pretty to rely on the slave to be 0, and the master >0, do you prefer : 
`if (LL_SPI_IsEnabled(spi) && (LL_SPI_GetMode(spi) ==  LL_SPI_MODE_MASTER)) {`

Reply:
> LL_SPI_IsEnabled return either 1 (true) or 0 (false)

Well, at this step, should always be 1, isn't it?

Reply:
It should, I had odd behaviors and errors with the driver, and I had to check if the SPI is still enabled or if we encountered an issue that disabled the SPI.
Now this is fixed, and we don't need to check if the SPI is enabled, so this part of the condition has been removed